doc/source/serve/dev-workflow.md

```diff
@@ -0,0 +1,123 @@
+% TODO(architkulkarni): Add in the appropriate sidebars and table of contents
+
+# Sample Development Workflow
+
+This page describes the recommended development workflow for Ray Serve applications when iterating in development. Once you're ready to go to production, you can jump to the [Production Guide](production-guide.md) section.
+
+## Project structure
+
+There are a few reasonable options for organizing your project structure, depending on the complexity of your application.  
+
+- You can put all of your code in a single Python file.
+- You can put your call graph in a Python file and your deployment class definitions in other files.
+
+## Local Development using Python handles
+
+You can use `serve.run` in process to run and test your application locally, using a Python handle to send requests rather than HTTP.
+
+Benefits:
+
+- Self-contained Python is convenient for writing local integration tests.
+- No need to deploy to a cloud provider or manage infrastructure.
+
+Drawbacks:
+
+- Doesn't test HTTP endpoints.
+- Doesn't test multi-node clusters, can't use GPUs if your local machine doesn't have them.
+
+Let's see a simple example.
+
+```python
+# Filename: local_dev.py
+from ray import serve
+from ray.dag.input_node import InputNode
+import starlette.requests
+
+@serve.deployment
+def double(s: str):
+    return s + " " + s
+
+@serve.deployment
+class HelloDeployment:
+    def __call__(self, request: starlette.requests.Request):
+        return "Hello, {}!".format(request.params["name"])
+
+# Everything beyond this line could be in a different file, provided
+# HelloDeployment and double are imported.
+hello_deployment = HelloDeployment.bind()
+with InputNode() as http_request:
+    hello_output = hello_deployment.bind(http_request)
+    double_output = double.bind(hello_output)
```

Comment:
When I run this I get an error about calling `bind.()` twice on a ClassNode

Reply:
I think this line:

```
hello_output = hello_deployment.bind(http_request)
```

should be this:

```
hello_output = hello_deployment.__call__.bind(http_request)
```

Could you try that and see if it fixes the issue?